### DISCUSSION TITLE
Is there a way to hoist a specific package version ?

### DISCUSSION THREAD

**Discussion by Exifers** [pnpm/pnpm#5332] [Q&A] [answered]
I'm trying to setup storybook in a pnpm monorepo. I'm facing [this issue](https://github.com/storybookjs/storybook/issues/18774) since apparently storybook requires `ts-node@10.9.1` but doesn't require it as a peer dependency.

I have other packages using `ts-node`: some use `ts-node@10.9.1` but others use `ts-node@9.1.1`.

Now since everything is hoisted by default, storybook finds ts-node in the root at `node_modules/.pnpm/node_modules/ts-node` but **it turns out it's always the oldest version (9.1.1) that is hoisted, not the latest one (10.9.1)**. Is there a way to specify what version is hoisted ?

<hr/>

I tried : 
```
hoist-pattern[]=ts-node@10.9.1
```
but it doesn't match the package (and nothing is hoisted then).

Thanks in advance !

**Comment by emosheeep** [comment, marked as answer]
Maybe you should add the corresponding peer dependencies to the problem packages. If you can't make some change，try to use .pmpmfile.cjs（readPackage hook） to fix peer problem.

**Reply by Exifers** [reply]
Ok I see, in the end I managed to fix it by changing the ts-node version in the other packages. But I didn't know about .pnpmfile.cjs it seems the right way fix this kind of problem.
Thanks for the answer.
